R/include-diabetes-diagnoses.R

```diff
@@ -0,0 +1,124 @@
+#' Include ICD-8 and ICD-10 Codes for Diabetes Diagnosis and Type-Specific Classification
+#'
+#' This function processes hospital contact data by combining outputs from `join_lpr2()` and `join_lpr3()` to identify diabetes-related diagnoses, evaluate inclusion criteria, and compute variables for type-specific classification.
+#'
+#' @param lpr_diag The diagnosis table from LPR2
+#' @param lpr_adm The admission table from LPR2
+#' @param diagnoser The diagnosis table from LPR3
+#' @param kontakter The contacts table from LPR3
+#'
+#' @return A tibble with the following columns:
+#' - `pnr`: Identifier variable (unique to individuals).
+#' - `date`: Dates of the first and second hospital diabetes diagnoses.
+#' - `any_t1d_primary_diagnosis`: Logical value indicating whether an individual has at least one type 1 diabetes-specific primary diagnosis code from either an endocrinological or other medical department.
+#' - `majority_t1d_primary_diagnoses`: Logical value indicating whether an individual has a majority of type 1 diabetes-specific primary diagnoses among all type-specific primary diagnoses recorded in endocrinological departments (or among diagnoses from other medical departments if no endocrinological contacts are recorded).
+#'
+#' The output is passed to `join_inclusions()`, where the `date` variable is used to define the inclusion date.
+#' The variables `any_t1d_primary_diagnosis` and `majority_t1d_primary_diagnoses` are subsequently passed to `get_diabetes_type()` for final classification of diabetes type.
+#'
+#' @keywords internal
+#'
+#' @examples
+#' \dontrun{
+#' lpr_diag <- simulate_registers("lpr_diag", 1000)[[1]]
+#' lpr_adm <- simulate_registers("lpr_adm", 1000)[[1]]
+#' diagnoser <- simulate_registers("diagnoser", 1000)[[1]]
+#' kontakter <- simulate_registers("kontakter", 1000)[[1]]
+#' include_diabetes_diagnoses(lpr_diag, lpr_adm, diagnoser, kontakter)
+#' }
+include_diabetes_diagnoses <- function(lpr_diag, lpr_adm, diagnoser, kontakter) {
```

Comment:
This should take two arguments: `lpr2` and `lpr3`, see https://github.com/steno-aarhus/osdc/blob/main/R/classify-diabetes.R#L86-L89